pkg/utils/gardener/deletion_confirmation.go

```diff
@@ -54,11 +54,14 @@ func CheckIfDeletionIsConfirmed(obj client.Object) error {
 }
 
 // ConfirmDeletion adds Gardener's deletion confirmation and timestamp annotation to the given object and sends a PATCH
-// request. It does not ignore `NotFound` errors while patching.
-func ConfirmDeletion(ctx context.Context, w client.Writer, obj client.Object) error {
+// request. If forceDelete is set to true, it adds the force deletion annotation as well. It does not ignore `NotFound` errors while patching.
+func ConfirmDeletion(ctx context.Context, w client.Writer, obj client.Object, forceDelete bool) error {
```

Comment:
This is called with a `true` argument only once in the test. I think @rfranzke's suggestion was more to having one utility function that takes as argument the annotation that has to be set (only the name of the annotation as the value is always `true`).
Then we could keep the `ConfirmDeletion` function and have it call this new utility function with the `ConfirmationDeletion` annotation as argument so as not to change the code in multiple places.

Reply:
If we are annotating with the force-delete and un-ignoring the Shoot in one patch call in the test, then we don't need this function at all.